plugins/inputs/logfile/logfilter.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package logfile
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/logs"
+	"regexp"
+)
+
+const (
+	includeFilterType = "include"
+	excludeFilterType = "exclude"
+)
+
+var validFilterTypesSet map[string]bool
+var validFilterTypes []string
+
+type LogFilter struct {
+	Type        string `toml:"type"`
+	Expression  string `toml:"expression"`
+	expressionP *regexp.Regexp
+}
+
+func (filter *LogFilter) init() error {
+	if validFilterTypes == nil {
+		validFilterTypes = []string{includeFilterType, excludeFilterType}
+		validFilterTypesSet = make(map[string]bool)
+		for _, f := range validFilterTypes {
+			validFilterTypesSet[f] = true
+		}
```

Comment:
You could do this right where `validFilterTypes` and `validFilterTypesSet` are declared, doesn't have to be inside this method.

Reply:
It didn't like me trying to loop through the splice to populate the map outside of a function definition. Not sure if what I have now is what you had in mind, but just copy pasting that code block out did not compile for me.

Reply:
Ah, sorry for recommending that without trying it out myself.  I like what you came up with.